pkg/datapath/loader/base.go

```diff
@@ -204,7 +204,8 @@ func (l *Loader) reinitializeIPSec(ctx context.Context) error {
 		if len(option.Config.IPv4PodSubnets) == 0 {
 			if info := node.GetRouterInfo(); info != nil {
 				for _, c := range info.GetIPv4CIDRs() {
-					option.Config.IPv4PodSubnets = append(option.Config.IPv4PodSubnets, &c)
+					cidr := c // create a copy to be able to take a reference
+					option.Config.IPv4PodSubnets = append(option.Config.IPv4PodSubnets, &cidr)
```

Comment:
I'm totally fine with the proposed patch, did you consider something like this to avoid creating a new variable on each loop?
```golang
				ipv4CIDRs := info.GetIPv4CIDRs()
				for i := range ipv4CIDRs {
					option.Config.IPv4PodSubnets = append(option.Config.IPv4PodSubnets, &ipv4CIDRs[i])
				}
```

Reply:
I have. I figured I make it more explicit with the copy to avoid someone accidentally "cleaning up" the logic, since this is already the second occurrence of this bug (in a branch of mine I did exactly the same mistake while iterating through `GetIPv4CIDRs`).